### DISCUSSION TITLE
How to disable SMTP in .env?

### DISCUSSION THREAD

**Discussion by iLikeToCode** [dani-garcia/vaultwarden#4866] [Q&A] [answered]
I’m running vaultwarden on the host, not docker, and for some reason no config.json is being created / admin page settings aren’t saving, and I want to disable SMTP via .env, but can’t find the tag for the .env file

**Comment by BlackDex** [comment, marked as answer]
Just do not configure any of the SMTP variables and you are fine.
Also, if there is no config.json it should just read all the items from the ENV.

**Reply by iLikeToCode** [reply]
Hopefully that works, let me test it

**Reply by iLikeToCode** [reply]
In the admin page, SMTP still has the enabled checkbox, it’s the default
Even then, how do I invite a user and let them sign in/up without smtp sending the link

**Reply by BlackDex** [reply]
That feature doesn't mind. But you can always uncheck that box, save and it will also be disabled.
That way you can keep your SMTP settings fully configured, but SMTP not working.

See:
https://github.com/dani-garcia/vaultwarden/blob/3466a8040e33c8ac359fed99f3542ed5ed6603fa/src/config.rs#L1225-L1228

**Reply by iLikeToCode** [reply]
Yeah, but I still can’t figure how to properly make a user account after inviting without an email coming through. If I should make a new question or an issue, I will, but I have registration off for a reason, worst case I could temp enable it, but that’s hassle.

**Comment by iLikeToCode** [comment]
I got invites working too, by setting up an smtp relay server on an infinityfree free domain, but invites seem to be missing functionality if no smtp server

**Reply by BlackDex** [reply]
What kind of functionality?? Without a mail server no invites can be aent by mail. And people just need to create an account with the invites mail address used to invite someone.

We can't magically mail people without a mail server.

**Reply by iLikeToCode** [reply]
Point is, the register backend page returns 404 if signups are off but invites are on, even when making account with invited email

**Reply by BlackDex** [reply]
Then something else is wrong. What do you mean with register backend page? What are the logs etc....

**Reply by iLikeToCode** [reply]
@BlackDex 
<img width="414" alt="image" src="https://github.com/user-attachments/assets/f54246b7-692a-4716-a80d-333612945d2e">


identity/accounts/register is a 404

Invites are on and signups are off

**Reply by iLikeToCode** [reply]
I did build everything myself so I could run it outside of docker

It’s not found even when signups are enabled

**Reply by BlackDex** [reply]
Rather looks like you did not configured the domain in a valid way. You seem to use a subpath so did  you configured `DOMAIN=https://my.domain.tld/vault` ? Because that is how it should be if you use subpaths.

Also reverse proxy sometimes need special config for subpaths so that those too.

**Reply by iLikeToCode** [reply]
I did all that, I was gonna follow #241 until I actually read it and figured you can do it in the DOMAIN, also nginx is configured, I did it cus I want

example.com front page
example.com/vault vault warden

**Reply by iLikeToCode** [reply]
The admin page is working, I’ll scroll through the rust code

**Reply by BlackDex** [reply]
I'm not sure what you are trying to do here.
Vaultwarden does not support a mixture of subpaths and no subpaths.
It's either one or the other.

So, if you want to access your vault via `https://my.domain.tld/vault` configure the `DOMAIN` like that, and no other rewriting on the reverse proxy side. Or if you want to access it via `https://vaultwarden.domain.tld` use that. Not a mixture.

The `/admin/diagnostics` page should show you if there is anything invalid configured (unless you do not access that via a subpath, but that is then a misconfiguration in general on your side).

**Reply by iLikeToCode** [reply]
not mixing, nginx handles it, VW is set to run at /vault
Checking diagnostics now

**Reply by iLikeToCode** [reply]
Could it be that I’m on 1.23.1 instead of 1.32.0

**Reply by BlackDex** [reply]
Double check your nginx config with the examples provided here: https://github.com/dani-garcia/vaultwarden/wiki/Proxy-examples See the `Nginx with sub-path (by @BlackDex)` example

**Reply by BlackDex** [reply]
v1.23.1 is very old and contains CVE issues. I suggest to upgrade a.s.a.p.

**Reply by iLikeToCode** [reply]
Yeah I used that one

**Reply by iLikeToCode** [reply]
I’ll pull the latest repo and build it

**Reply by BlackDex** [reply]
To explain, you are probably still using `bitwardenrs/server*` as your docker image.
Convert that to `vaultwarden/server`

**Reply by iLikeToCode** [reply]
Already said, not using a docker image, I built it manually and am running it with systemd

**Reply by iLikeToCode** [reply]
Ah, I was using the new web vault and old executable, path must have changed between versions, it works now

**Comment by iLikeToCode** [comment]
I’ll respond l8r, I’m on holiday

But I have signups off so the register page was off
So front end was on but
/ something / register was disabled
